src/features/todos.ts

```diff
@@ -1,9 +1,24 @@
 import { Todo } from '../types/Todo';
 
-export const actions = {};
+type Actions = {
+  type: 'todos/ADD';
+  payload: Todo[];
+};
+
+const todosAdd = (todos: Todo[]): Actions => ({
+  type: 'todos/ADD',
+  payload: todos,
+});
+
+export const actions = { todosAdd };
 
-const todosReducer = (): Todo[] => {
-  return [];
+const todosReducer = (todos: Todo[] = [], action: Actions): Todo[] => {
```

Comment:
```suggestion
const todosReducer = (todos: Todo[] = [], action: Actions): Todo[] => {
```
Reducer should take state as a Param State should be an object

Reply:
Also you can hold isLoading or/and error state